DQL custom functions: document TypedExpression (#11550)
doctrine/doctrine2: docs/en/cookbook/dql-user-defined-functions.rst

process, for example if the Unit would not be one of the supported
values by MySql.

Typed functions

> ----------

Reviewer: ```suggestion
Typed functions
---------------
```

By default, result of custom functions is fetched as-is from the database driver.

> If you want to be sure the type is always the same, your custom function can 

Reviewer: ```suggestion
If you want to be sure that the type is always the same, then your custom function needs to 
```

implement ``Doctrine\ORM\Query\AST\TypedExpression``. Then, the result is wired

> through ``convertToPhpValue`` method of the ``Type`` returned from ``getReturnType``.

Reviewer: ```suggestion
through the ``Doctrine\DBAL\Types\Type::convertToPhpValue()`` method returned in ``getReturnType``.
```

Author: This suggestion feels more weird to me. No method is returned (obviously), Type is returned.

Reviewer: I didn't expect that someone could read that a "method" is returned.

```suggestion
through ``Doctrine\DBAL\Types\Type::convertToPhpValue()`` of the ``Type`` returned in ``getReturnType()``.
```
